pkg/tlsutil/tls.go

```diff
@@ -188,16 +196,54 @@ func (scg *SDKCertGenerator) GenerateCert(cr runtime.Object, service *v1.Service
 		}
 		return appSecret, caConfigMap, caSecret, nil
 	} else {
+		var caKeyData, caCertData []byte
+		var err error
+		var caKey *rsa.PrivateKey
+		var caCert *x509.Certificate
 		// case: both CA and Application TLS assets don't exist.
-		caKey, err := newPrivateKey()
-		if err != nil {
-			return nil, nil, nil, err
+		if config.CAKey != "" && config.CACert != "" {
```

Comment:
I think this chain of `if` and `if/else` could be clearer if in a switch? Thoughts?

Reply:
I switched this from an if/else to a switch cause I thought it would be cleared. You would just have to look at the conditional statement as opposed to calculate which case is remaining